ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts/vlans.py

```diff
@@ -46,16 +46,21 @@ def _parse_adapter_settings(self: EosDesignsFacts, adapter_settings: dict) -> tu
         trunk_groups = set(adapter_settings.get("trunk_groups", []))
         if "vlans" in adapter_settings and adapter_settings["vlans"] not in ["all", "", None]:
             vlans.update(map(int, range_expand(str(adapter_settings["vlans"]))))
-        elif "trunk" in adapter_settings.get("mode", "") and not trunk_groups:
+        elif adapter_settings.get("mode", "") == "trunk" and not trunk_groups:
             # No vlans or trunk_groups defined, but this is a trunk, so default is all vlans allowed
             # No need to check further, since the list is now containing all vlans.
             return set(range(1, 4094)), trunk_groups
+        elif adapter_settings.get("mode", "") == "trunk phone":
+            # Phone trunk, leave vlans empty
```

Comment:
Please add some more details here, why is this the right thing?
The `vlans` key is still not set, so we need to configure the data vlan.

Reply:
All the examples for phone VLAN (in [this TOI](https://www.arista.com/en/support/toi/eos-4-28-2f/16061-phone-vlan)) are either normal trunk ports, or when in `phone trunk` mode, only define the native and phone vlan.  The TOI states (and this can be seen from the CLI) that only the phone and native vlans are configured on the trunk.
```
bri028(config-if-Et53/1)#sh vl br
VLAN  Name                             Status    Ports
----- -------------------------------- --------- -------------------------------
1     default                          active
10    VLAN0010                         active    Cpu, Et53/1
20    VLAN0020                         active    Cpu, Et53/1
30    VLAN0030                         active    Cpu

bri028(config-if-Et53/1)#sh act
interface Ethernet53/1
   no shutdown
   switchport trunk native vlan 10
   switchport phone vlan 20
   switchport mode trunk phone
bri028(config-if-Et53/1)#sh int tru
There are no active trunk ports
bri028(config-if-Et53/1)#sh int tru phone
Port            Mode            Status          Native vlan
Et53/1          trunk phone     trunking        10

Port            Vlans allowed
Et53/1          All

Port            Vlans allowed and active in management domain
Et53/1          10,20

Port            Vlans in spanning tree forwarding state
Et53/1          10,20
```
We (internally in AVD) add them to the vlans list for automatic trunking on l2 uplinks, lines 60-63 below.

Reply:
If so, we should either guard to error if native vlan is missing or specifically check for the native vlan with a default of 1 (eos default), so we still add vlan 1.